### ISSUE TITLE
Unexpected errors from contract checking for !null -> !null contracts

### ISSUE THREAD

**Issue by wilkinsona** [uber/NullAway#1440]
Using NullAway 0.13.0, we're seeing unexpected errors reported when contract checking is enabled. It seems to be related to `!null` contracts such as the following three variants of the same method:

```java
@Contract("!null -> !null")
public static @Nullable Integer of(@Nullable String text) {
	if (text != null) {
		return Integer.parseInt(text);
	}
	else {
		return null;
	}
}
```

```java
@Contract("!null -> !null")
public static @Nullable Integer of(@Nullable String text) {
	if (text != null) {
		return Integer.parseInt(text);
	}
	return null;
}
```

```java
@Contract("!null -> !null")
public static @Nullable Integer of(@Nullable String text) {
	return (text != null) ? Integer.parseInt(text) : null;
}
```

FWIW, IntelliJ's null checker reports no problems with any of these.

If it's useful, you can see more examples of some unexpected failures in https://github.com/spring-gradle-plugins/nullability-plugin/issues/10.

**Comment by giruuuuj** [comment]
## Problem
NullAway was incorrectly reporting contract violations for methods with `@Contract("!null -> !null")` when they returned null in conditional branches that check if the parameter is null.

## Solution
Modified [ContractCheckHandler](cci:2://file:///e:/gssoc%20project/NullAway/nullaway/src/main/java/com/uber/nullaway/handlers/contract/ContractCheckHandler.java:56:0-358:1) to check if return statements are reachable when antecedent conditions are met. The fix adds AST analysis to detect when return statements are inside conditional blocks checking `param == null`.

## Changes
- Enhanced [ContractCheckHandler.java](cci:7://file:///e:/gssoc%20project/NullAway/nullaway/src/main/java/com/uber/nullaway/handlers/contract/ContractCheckHandler.java:0:0-0:0) with reachability analysis
- Added comprehensive test cases in [ContractsIssue1440Test.java](cci:7://file:///e:/gssoc%20project/NullAway/nullaway/src/test/java/com/uber/nullaway/ContractsIssue1440Test.java:0:0-0:0)
- All existing tests continue to pass

## Testing
- Verified that the three patterns from issue #1440 no longer trigger false positives
- Confirmed that actual contract violations are still properly detected
- All existing contract tests pass

Fixes #1440

**Comment by msridhar** [comment]
Thanks so much for testing, @wilkinsona!  I think the gap was in handling of returned ternary expressions.  I've updated #1441 to fix those now.  Any chance you could give it another shot?
